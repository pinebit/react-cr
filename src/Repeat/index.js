import React from 'react';
import Wrapper from '../Wrapper';

const Repeat = ({ count, children, ...wrapperProps }) => {
  if (!children || count <= 0 || Array.isArray(children)) {
    return null;
  }

  if (count === 1) {
    return (
      <Wrapper {...wrapperProps}>
        {children}
      </Wrapper>
    );
  }

  const fakeArray = Array.from(new Array(count)).map((x, index) => index + 1);

  return (
    <Wrapper {...wrapperProps}>
      {fakeArray.map(key => React.cloneElement(children, { key }))}
    </Wrapper>
  );
};

Repeat.propTypes = {
  count: React.PropTypes.number.isRequired,
  children: React.PropTypes.node
};

export default Repeat;
