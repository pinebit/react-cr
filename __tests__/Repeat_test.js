import React from 'react';
import renderer from 'react-test-renderer';
import Repeat from '../src/Repeat';

it('Repeat renders children 3 times', () => {
  const tree = renderer.create(
    <Repeat count={3}>
      <h1>1</h1>
    </Repeat>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Repeat does not render when count is zero or negative', () => {
  const tree = renderer.create(
    <Repeat count={0}>
      <h1>1</h1>
    </Repeat>
  ).toJSON();

  expect(tree).toMatchSnapshot();

  const tree2 = renderer.create(
    <Repeat count={-1}>
      <h1>1</h1>
    </Repeat>
  ).toJSON();

  expect(tree2).toMatchSnapshot();
});

it('Repeat does not render when children are array', () => {
  const tree = renderer.create(
    <Repeat count={3}>
      <h1>1</h1>
      <h1>2</h1>
    </Repeat>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Repeat does not wrap when count is one', () => {
  const tree = renderer.create(
    <Repeat count={1}>
      <h1>1</h1>
    </Repeat>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
