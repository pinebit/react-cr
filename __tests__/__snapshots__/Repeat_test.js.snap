// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repeat does not render when children are array 1`] = `null`;

exports[`Repeat does not render when count is zero or negative 1`] = `null`;

exports[`Repeat does not render when count is zero or negative 2`] = `null`;

exports[`Repeat does not wrap when count is one 1`] = `
<h1>
  1
</h1>
`;

exports[`Repeat renders children 3 times 1`] = `
<div>
  <h1>
    1
  </h1>
  <h1>
    1
  </h1>
  <h1>
    1
  </h1>
</div>
`;
